src/Lookup/InProcessCachingDataTypeLookup.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Lookup;
+
+use Wikibase\DataModel\Entity\PropertyId;
+
+/**
+ * PropertyDataTypeLookup that provides in-process caching.
+ *
+ * @since 3.1
+ *
+ * @licence GNU GPL v2+
+ * @author Katie Filbert < aude.wiki@gmail.com >
+ */
+class InProcessCachingDataTypeLookup implements PropertyDataTypeLookup {
```

Comment:
Wow, so names **do** matter a lot. By typing "InMemory" I instantly found an existing [InMemoryDataTypeLookup](https://github.com/wmde/WikibaseDataModelServices/blob/master/src/Lookup/InMemoryDataTypeLookup.php) class that already does what this proposal does.